clang/include/clang/Driver/Options.td

```diff
@@ -7209,6 +7209,9 @@ def fuse_ctor_homing: Flag<["-"], "fuse-ctor-homing">,
 def as_secure_log_file : Separate<["-"], "as-secure-log-file">,
   HelpText<"Emit .secure_log_unique directives to this filename.">,
   MarshallingInfoString<CodeGenOpts<"AsSecureLogFile">>;
+def output_asm_variant : Joined<["--"], "output-asm-variant=">,
+  HelpText<"Select the asm variant (integer) to use for output">,
```

Comment:
Requiring the user to pick a variant via an integer index feels a bit user-unfriendly - should/could we expose this with names like `=intel` and `=att` like we do for the other related options? Admittedly, this is not a very user facing option, as it is a`-cc1` option only, if I'm correctly understanding the situation, but nevertheless, I wouldn't offhand have any idea about what indices to use here.

Reply:
Agreed with the conservation. The ergonomics is one the reasons that this option is kept internal (cl::opt, cc1, not driver).

llvm-mc and llc currently use output-asm-variant=1 and the integer looks better for non-x86 targets.

Reply:
For a GNU inline asm statement, it's possible to specify multiple dialects, like in

```
__asm__ ("{ movl $1, %eax | mov eax, 1 }");
```

so index zero is AT&T and index one is Intel, hard-coded in source code like that.

A numeric argument is probably not too bad if it matches another tool.

Reply:
Thanks for the GCC information. Looks good that the numerical index, already used by existing internal LLVM tools, also matches GCC internals.

Reply:
Appreciate the comments. 

Yes, the AssemblerDIalect choices could be useful for other targets. SystemZ has `AssemblerDIalect`s that I am not familiar with. AArch64 seems to have one as well.

Reply:
Oh, indeed - yes, there's a nonstandard dialect for NEON instructions in aarch64, which is used for disassembly on Darwin targets by default. Thanks for the pointers!